src/quadratic_polyline2_cost.cpp

```diff
@@ -55,9 +55,13 @@ float QuadraticPolyline2Cost::Evaluate(const VectorXf& input) const {
 
   // Compute signed squared distance by finding closest point.
   float signed_squared_distance;
+  bool is_endpoint;
   polyline_.ClosestPoint(Point2(input(xidx_), input(yidx_)), nullptr, nullptr,
-                         &signed_squared_distance);
-
+                         &signed_squared_distance, &is_endpoint);
+  if(is_endpoint){
+    // endpoint fix
+  	signed_squared_distance = 0;
+  }
   return 0.5 * weight_ * std::abs(signed_squared_distance);
```

Comment:
nit: blank line before

Reply:
done!

Reply:
See comment above about not seeing change.

Reply:
On second thought, you can just disregard this comment